### PR TITLE
feat: No Assessment Table Component

### PR DESCRIPTION
**Pull request by JennaXiao3**
## Notion ticket link

[No Assessments table component](https://www.notion.so/uwblueprintexecs/No-Assessments-table-component-3033888b0c4f4537af2519085c7c0785?pvs=4)

## Implementation description
* created a component for when there are no assessments

## Steps to test
1. test in [localhost](http://localhost:3000/admin/assessments)

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx [hunk 1]
```diff
@@ -82,19 +84,23 @@ const DisplayAssessmentsPage = (): React.ReactElement => {
...
     let filteredTests: AssessmentProperties[] = data.tests as AssessmentProperties[];
-    const filterProps = [grades, testTypes, countries, regions, status];
+    const filterProps = [status, grades, testTypes, countries, regions];
 
     filterProps.forEach((property, i) => {
       filteredTests = filteredTests.filter(
         (assessment: AssessmentProperties) => {
+          if (i === 1) setEmpty(false);
           const assessmentProperties = [
+            assessment.status,
```
Comment (joyce-shi) [inline frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx]: do we need to switch the order for `assessment.status`?

File: frontend/src/components/common/table/NoResultsTableState.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <Text paddingBottom="2rem" textStyle="paragraph">
+          Create your first assessment
+        </Text>
+        <Button variant="primary">Create assessment</Button>
+      </VStack>
+    </Center>
+  );
+};
+
+export default NoResultsTableState;
```
Comment (joyce-shi) [inline frontend/src/components/common/table/NoResultsTableState.tsx:29]: can we swap the naming here? I think the existing one should be called `EmptyTableState` and the other one should be `NoResultsTableState` based on their use case

File: frontend/src/components/common/table/NoResultsTableState.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          height="8rem"
+          paddingBottom="1rem"
+          src={DisplayAssessments}
+          width="8rem"
+        />
+        <Text textStyle="subtitle1">You currently have no assessments.</Text>
+        <Text paddingBottom="2rem" textStyle="paragraph">
+          Create your first assessment
+        </Text>
+        <Button variant="primary">Create assessment</Button>
```
Comment (joyce-shi) [inline frontend/src/components/common/table/NoResultsTableState.tsx]: for this button, can we add functionality to it? I think it should redirect to the **Create Assessment** page (same behaviour as the other Create Assessment button on this page

**Review by joyce-shi — COMMENTED** [review]
loooks awesome!! left some small comments! a couple additional notes:

a. i think we only want this state to show when there are no items in the table at all (so that **Create your first assessment** makes sense). given this, i think `isEmpty` should only be true when there are no assessments returned from the **GET_ALL_ASSESSMENTS** call. right now, this state shows based filters (for example, if you remove all Archived assessments and then navigate to the Archived tab you can see this state). would we be able to update this behaviour?

b. we have an existing component that visually looks very similar to your implementation here, called `MessageContainer`. would you be able to see if you can refactor your implementation to use this general component? feel free to make updates to clean up `MessageContainer` a bit! an example for where this is already used is `AddAdminConfirmationMessage`
For context, in the next sprint, we also want the following designs will be able to use this `MessageContainer` component!
- [Mobile Redirect](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=3854-94331&t=kIMIG02FktFCH3s6-4)
- [Test Submission Message](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=3475-90969&t=kIMIG02FktFCH3s6-4)

**Comment by JennaXiao3** [comment]
> 

hey @joyce-shi ! 

a. in the ticket it was mentioned that "This screen should only show when there are no assessments at all **under that tab category**." Was this a mistake then? So only when we first get started?

File: frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx (hunk 1 above)
Comment (JennaXiao3) [reply frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx]: the reason for this was cause i misunderstood the ticket - i thought it was when there are no assessments under a tab catagory and not in general

**Comment by joyce-shi** [comment]
> Member

oops, that's probably my bad. would you be able to double check with design what the intended behaviour is? i think bonnie worked on it. sorry for all the back and forth!

**Comment by JennaXiao3** [comment]
sorry for the delay!!!

**Review by joyce-shi — APPROVED** [review]
LGTM!

**Pull request merged**